tests/trainer/test_dataloaders.py

```diff
@@ -1470,35 +1471,28 @@ def __iter__(self):
         def __next__(self):
             return next(self._iter)
 
-    class DataLoaderFunc:
-        def __init__(self, loader):
-            self.loader = loader
-
-        def __call__(self):
-            return self.loader
-
     class TestModel(BoringModel):
         def __init__(self):
             super().__init__()
-            self.on_train_dataloader_called = False
+            self.train_dataloader_called = False
             self.on_train_batch_start_called = False
-            self.on_val_dataloader_called = False
+            self.val_dataloader_called = False
             self.on_val_batch_start_called = False
 
-        def on_train_dataloader(self) -> None:
-            loader = self.train_dataloader()
-            self.train_dataloader = DataLoaderFunc(DataLoaderWrapper(loader))
-            self.on_train_dataloader_called = True
+        def train_dataloader(self) -> TRAIN_DATALOADERS:
+            loader = super().train_dataloader()
+            self.train_dataloader_called = True
+            return DataLoaderWrapper(loader)
+
+        def val_dataloader(self) -> EVAL_DATALOADERS:
+            loader = super().val_dataloader()
+            self.val_dataloader_called = True
+            return DataLoaderWrapper(loader)
 
         def on_train_batch_start(self, batch, batch_idx: int, dataloader_idx: int) -> None:
             assert isinstance(self.trainer.train_dataloader.loaders, DataLoaderWrapper)
             self.on_train_batch_start_called = True
 
-        def on_val_dataloader(self) -> None:
-            loader = self.val_dataloader()
-            self.val_dataloader = DataLoaderFunc(DataLoaderWrapper(loader))
-            self.on_val_dataloader_called = True
-
```

Comment:
IMO, you could split this out to a separate PR after the deprecation. That will confirm that the initial deprecation is backwards compatible & safe, and the cleanup afterwards demonstrates how to migrate to the supported API